packages/neptune-extension/src/common/components/upload-modal/UploadModal.tsx

```diff
@@ -78,14 +81,19 @@ const UploadModal: React.FC<UploadModalProps> = ({
   // By default always try to create new checkpoint
   const [ mode, setMode ] = React.useState<UploadMode>(canUploadCheckpoint ? 'checkpoint' : 'notebook');
 
-  const [ projectId, setProjectId ] = React.useState(() => {
-    return !!notebook
-      ? notebook.projectId
-      : window.localStorage.getItem(PROJECT_LOCAL_STORAGE_KEY) || ''
-  });
+  const defaultProjectId = !!notebook
+    ? notebook.projectId
+    : window.localStorage.getItem(PROJECT_LOCAL_STORAGE_KEY) || '';
+
   const [ name, setName ] = React.useState('');
   const [ description, setDescription ] = React.useState('');
 
+  const [ projectId, projectInputProps, projectMetaProps, setProjectId ] = useSelectInputValue(
```

Comment:
Here you should use `projectInputProps.onChange` and not change the interface.

Reply:
Ok